src/gt4py/gtc/gtcpp/gtcpp.py

```diff
@@ -158,11 +159,32 @@ class GTFunctor(LocNode, SymbolTableTrait):
     param_list: GTParamList
 
 
-# A ParamArg is an argument that maps to a parameter of something with the same name.
-# Because all things are called exactly once there is a one-to-one mapping.
-# TODO with symbol table the concept probably doesn't make sense anymore
-class ParamArg(LocNode):
-    name: Str
+class Param(LocNode):
+    name: SymbolName
+
+    class Config(eve.concepts.FrozenModelConfig):
+        pass
+
+    # TODO see https://github.com/eth-cscs/eve_toolchain/issues/40
```

Comment:
Btw, since you are linking to this from here, it would be nice if that issue had some quick explanation how it leads to this code pattern currently.

Reply:
done